### ISSUE TITLE
Make sure there is a loader supporting the "api_platform" type. (how to interpret)

### ISSUE THREAD

**Issue by cjchamado** [api-platform/api-platform#1880]
**Scenario**
Operating System: macOS 11.3 (Big Sur)
PHP version: PHP 8.0.2 (cli) (built: Feb 4 2021 18:57:55) (NTS)

**Exception**
could not find driver in. (which is being imported from "/...project-dir.../config/routes/api_platform.yaml"). Make sure there is a loader supporting the "api_platform" type.

**Observation**
What can be noticed is a small confusion of interpretation that occurs when ApiPlatform loads routes.

When the entities metadata loading class is triggered, Doctrine ORM tries to make a connection to the database (the reason for this is not described here), and, if your development environment does not have the bank connection driver of data, an exception is thrown by the PDO.

Exception message: could not find driver

As the exception above occurred in the context of loading files that have the routing settings, this ends up finally being interpreted in ```Symfony\Component\Config\Loader\FileLoader::doImport(...) L:173```. In this excerpt there is a treatment that returns the corresponding exception "LoaderLoadException", if your instance is compatible. Otherwise, an instance of "LoaderLoadException" is created from the original exception (eg: PDOException).

At this point ```Symfony\Component\Config\Exception\LoaderLoadException::__ construct```, the message is "constructed" and, because there is a preceding exception, the message is prefixed with the message from the previous exception, generating the message a little confused, quoted at the beginning.

In short: Although the exception thrown gives the impression that the error is in the routing configuration that involves the "api_platform" type, it actually concerns the fact that your environment does not have a database connection driver (in this case).

Observing that ```ApiPlatform\Core\Bridge\Symfony\Routing\ApiLoader::load``` does not do any kind of treatment that isolates possible exceptions at this point, we can conclude that many errors (from other points of the application) that are launched at this point, will finally be treated as route loading error and suffixed with "Make sure there is a loader supporting the" api_platform "type.", it is in this case to investigate the message that prefixes this (eg: could not find driver).

**Comment by soyuka** [comment]
Closing as stale per #2987 (no activity since 2021-04-17). This repository is transitioning to a CLI installer. If this still reproduces against a current release of API Platform, please open a new issue on `api-platform/core` with a minimal reproduction.
